platform/core/PolyPodCoreAndroid/core/src/main/java/coop/polypoly/core/CoreExceptions.kt

```diff
@@ -0,0 +1,30 @@
+package coop.polypoly.core
+
+import Failure
+
+@ExperimentalUnsignedTypes
+class InternalCoreException(message: String) : Exception(message) {
+    companion object {
+        fun make(context: String, failure: Failure): InternalCoreException {
+            return InternalCoreException(
+                "$context -> internal Core Failure: ${failure.code} ${failure.message}"
+            )
+        }
+    }
+}
+
+class InvalidResultException(message: String) : Exception(message) {
+    companion object {
+        fun make(context: String, result: String): InvalidResultException {
+            return InvalidResultException(
+                "$context -> received invalid result type: $result"
+            )
+        }
+    }
+}
+
+class InvalidFailureContentException(context: String) :
+    Exception("$context -> received failure result type without content")
```

Comment:
"without" or "with invalid"?

Reply:
It appears "without" is accurate - this is being used above when the whole `failure` object is `null`.

Reply:
The thing is that the class is actually called "Invalid". So I guess it's more of a naming nit.

Reply:
True,`MissingFailureContentException` would be a better name.